internal/controller/backupconfig_controller.go

```diff
@@ -48,9 +53,60 @@ type BackupConfigReconciler struct {
 func (r *BackupConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	logger := logr.FromContextOrDiscard(ctx)
 	logger.V(1).Info("Running Reconcile for BackupConfig")
+
+	backupConfig := &v1beta1.BackupConfig{}
+	if err := r.Get(ctx, req.NamespacedName, backupConfig); err != nil {
+		// The Backup resource may have been deleted
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
 	// TODO(endevir): Check && report connection status into Status subresource
 	// TODO(endevir): Watch for secrets name change and reconcile cluster role
 
+	if !backupConfig.DeletionTimestamp.IsZero() {
```

Comment:
Can you pls implement tests on such behavior in `internal/controller/backupconfig_controller_test.go` ?
It will be useful to make auto-checks on that we keep finalizer until all backups owned by this BackupConfig are removed